x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/JdbcResultSet.java

```diff
@@ -934,7 +935,11 @@ public Clob getClob(int columnIndex) throws SQLException {
 
     @Override
     public Array getArray(int columnIndex) throws SQLException {
-        throw new SQLFeatureNotSupportedException("Array not supported");
+        EsType type = columnType(columnIndex);
+        if (isArray(type) == false) {
+            throw new SQLException("Cannot get an array for the column [" + columnIndex + "] of type [" + type.getName() + "]");
```

Comment:
Maybe `Cannot get column ... of type ... as array`?

Reply:
Better, fixed, thx.